Feature/aqi 9220

AQI-DanG opened the pull request.

MatthewDeRose-AI commented on src/ONE.Test.ConsoleApp/Commands/I18nCommand.cs:
```diff
@@ -0,0 +1,29 @@
+		async Task<int> ICommand.Execute(ClientSDK.OneApi clientSdk)
+		{
+			var results = await clientSdk.Library.Geti18nKeysAsync("en-US", "AQI_FOUNDATION_LIBRARY");
+			//var results = await clientSDK.Library.GetMobilei18nKeysAsync("en-US");
```
Can this be removed?

AQI-DanG replied:
Perhaps I should have added a comment to this effect but the ConsoleApp project is a half finished attempt at creating a CLI for the SDK

MatthewDeRose-AI commented on src/ONE.Test.ConsoleApp/Commands/CommandHelper.cs:
```diff
@@ -0,0 +1,121 @@
+		{
+			var configuration = ConfigurationManager.OpenExeConfiguration(ConfigurationUserLevel.None);
+			var accessToken = "";
+			if (configuration.AppSettings.Settings["AccessToken"] != null)
```
This is a big PR so I'm not sure if I missed something but if you reference these strings anywhere else you may want to centralize them as constants

MatthewDeRose-AI commented on src/ONE.ClientSDK/Utilities/RestHelper.cs:
```diff
@@ -0,0 +1,676 @@
+
+				Event(null,
+					new ClientApiLoggerEventArgs
+						{ File = file, EventLevel = eventLevel, HttpStatusCode = response.StatusCode, ElapsedMs = elapsedMs, Module = "RestHelper", Message = message });
```
I see POST and RestHelper are common hard coded strings in this file, it would be good to make them into centralized constants

MatthewDeRose-AI replied:
For any comments about centralizing strings I see that there is a lot of work in this PR. If needed it might be good to just create a tech debt task for it, just to improve maintainability

AQI-DanG replied:
I replaced any usages of RestHelper within the SDK with the OneApiHelper.  This allows those classes to be unit testable as the OneApiHelper can be mocked.  I left this class in place because it is a public interface for the SDK and I don't know if any consumers might be using it directly.  I would prefer to remove it entirely but that sort of change would probably need it's own epic.

MatthewDeRose-AI commented on src/ONE.ClientSDK/Utilities/RestHelper.cs:
```diff
@@ -0,0 +1,676 @@
+				form.Add(fileContent, "file", Path.GetFileName(filePath));
+				//form.Add(new StringContent("789"), "userId");
+				//form.Add(new StringContent("some comments"), "comment");
+				//form.Add(new StringContent("true"), "isPrimary");
```
If these commented lines aren't needed anymore please remove them.

MatthewDeRose-AI commented on src/ONE.ClientSDK/Enterprise/Report/ReportApi.cs:
```diff
@@ -0,0 +1,357 @@
+		{
+			try
+			{
+				var endpoint = $"enterprise/report/v1/report/output/{id}/template?requestId={Guid.NewGuid()}";
```
I can create the tech debt task if you're busy right now

AQI-DanG replied:
Yea, I noticed that the SDK only supports 'reports/v1', if we need support for other versions we would need to add that when the need becomes apparent.  I suspect that these methods are not actually used currently and I cannot guarantee they actually work

MatthewDeRose-AI reviewed (approved):
This is a pretty immense PR, my review is probably a bit superficial. Any comments above can be tech debt tasks to improve maintainability if timing is tight (unless otherwise specified).

AQI-DanG commented:
> This is a pretty immense PR, my review is probably a bit superficial. Any comments above can be tech debt tasks to improve maintainability if timing is tight (unless otherwise specified).

Yes it is and I apologize for that.  The whitespace settings on my new computer were screwed up and when I changed the namespaces it replaced all the indentation tabs with spaces.  I went back and changed them back to tabs but it seems like it thinks I changed / replaced almost every file.

Thank you for the review

KentMiller reviewed (approved):
It's hard to review since. All of the files were moved so everything looks like newly added code
